### PR TITLE
Bump Play to 2.8

### PR DESCRIPTION
**Pull request by jonathonherbert**
~**NB:** Depends on https://github.com/guardian/editorial-tools-platform/pull/646. Don't merge until this is in place – the app will not start!~ Now sorted.

## What does this change?

Bumps Play to 2.8, and pan-domain-auth to 1.2.0.

Skips Play 2.6, as it requires a different pan-domain-auth setup.

## How to test

Run locally, or in CODE. Browse, create, and modify story packages. Does everything work as expected?

~NB: Because we've recreated the story-packages stack recently, the old packages currently served by CAPI do not work! Additionally, because (I think!) the story-packages stream is not being read by CODE CAPI, new updates cannot be searched. So the only functionality we can currently test is creating a new story-package and editing that.~

~So at the moment, we can prove that~
 - ~we can talk to CAPI~
 - ~we can read and write packages from DynamoDB~
 - ~we can write updates on a stream (I've tested this locally)~

~But we cannot verify the search-for-package / select-package behaviour.~

CAPI's stream-reading behaviour is now sorted. I've made one behavioural change in [1500599](https://github.com/guardian/story-packages/pull/174/commits/15005996b7c0da5e0ed6baab27f2518115c74562) to make sure that the app does not 500 when CAPI surfaces content that is not in story-package's DB – instead, we just filter out the packages that we don't recognise. That seems less disruptive for the user – we still log the error message as normal. As a result, search should now work as well.

### CODE REVIEW

File: app/conf/Configuration.scala [hunk 1]
```diff
@@ -73,22 +85,26 @@ class ApplicationConfiguration(val playConfiguration: PlayConfiguration, val env
           None
       }
     }
+
+    val s3Client: Option[AmazonS3] = credentials.map { credentials =>
```
Comment (jonathonherbert) [inline app/conf/Configuration.scala:89]: Moved into config from the S3 trait to share with pan-domain-auth.

**Review by davidfurey — APPROVED** [review]
Minor point about the `lazy val` but otherwise looks good to me

File: app/controllers/StoryPackagesController.scala [hunk 2]
```diff
@@ -80,10 +80,13 @@ class StoryPackagesController(val config: ApplicationConfiguration, database: Da
       val json: JsValue = Json.parse(response.body)
       val packageIds = (json \ "response" \ "results" \\ "packageId").map(_.as[String])
       for {
-        packages <- Future.sequence(packageIds.map(id => database.getPackage(id)))
+        packages <- Future.sequence(packageIds.map(id => database.getPackage(id).transformWith {
```
[the remark this answers is not in the document]
Comment (jonathonherbert) [reply app/controllers/StoryPackagesController.scala]: Happy to use that, it's a bit more concise! [6875631](https://github.com/guardian/story-packages/pull/174/commits/6875631e0213eda9d474a6b7a748d6c0328359ba)

File: app/conf/Configuration.scala [hunk 3]
```diff
@@ -49,7 +55,13 @@ class ApplicationConfiguration(val playConfiguration: PlayConfiguration, val env
   object aws {
     lazy val region = getMandatoryString("aws.region")
     lazy val bucket = getMandatoryString("aws.bucket")
-    lazy val crossAccount = getMandatoryBoolean("aws.crossAccount")
+
+    object endpoints {
+      private lazy val _region = Region.getRegion(Regions.fromName(region))
```
[the remark this answers is not in the document]
Comment (jonathonherbert) [reply app/conf/Configuration.scala]: `lazy jon`. I don't really know why any of the vars in this file are lazy. Corrected this instance in [360077a](https://github.com/guardian/story-packages/pull/174/commits/360077a7d2985761c9f9399b3c6d9c6a587106e5). Arguably better to remove the lot.

[the remark this answers is not in the document]
Comment (davidfurey) [reply app/conf/Configuration.scala]: Maybe one for a follow up PR

**Comment by davidfurey** [comment]
🥳

**Pull request merged**